src/firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.c

```diff
@@ -0,0 +1,70 @@
+#include "firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.h"
+
+#include <stdlib.h>
+
+typedef struct AllegroA3931MotorDriver
+{
+    PwmPin_t* pwm_pin;
+    GpioPin_t* reset_pin;
+    GpioPin_t* coast_pin;
+    GpioPin_t* mode_pin;
+    GpioPin_t* direction_pin;
+    GpioPin_t* brake_pin;
+    GpioPin_t* esf_pin;
+} AllegroA3931MotorDriver_t;
+
+AllegroA3931MotorDriver_t* io_allegro_a3931_motor_driver_create(
+    PwmPin_t* pwm_pin, GpioPin_t* reset_pin, GpioPin_t* coast_pin, GpioPin_t* mode_pin,
+    GpioPin_t* direction_pin, GpioPin_t* brake_pin, GpioPin_t* esf_pin)
+{
+    AllegroA3931MotorDriver_t* motor =
```

Comment:
Do we want to distinguish between 'motor' and 'motor driver' here?

Reply:
Good catch, meant to change that. That was carryover from an old version of the code.